### PR TITLE
Modulab-15061. Combo with favourite option

### PR DESCRIPTION
**Pull request by VictorDeBlas**
Favourite option to combos.

### CODE REVIEW

File: icons/icomoon.svg [hunk 1]
```diff
@@ -45,13 +45,13 @@
...
 <glyph unicode="&#xe925;" glyph-name="wifi" d="M214 384.667c166 164 432 164 596 0l-84-86c-118 118-310 118-428 0zM384 212.667c70 70 186 70 256 0l-128-128zM42 554.667c260 258 682 258 940 0l-86-86c-212 212-556 212-768 0z" />
-<glyph unicode="&#xe926;" glyph-name="toggle-off" horiz-adv-x="1152" d="M768 832h-384c-212.078 0-384-171.922-384-384s171.922-384 384-384h384c212.078 0 384 171.922 384 384s-171.922 384-384 384zM128 448c0 141.482 114.498 256 256 256 141.482 0 256-114.498 256-256 0-141.482-114.498-256-256-256-141.482 0-256 114.498-256 256zM768 192h-97.81c130.434 145.716 130.472 366.24 0 512h97.81c141.482 0 256-114.498 256-256 0-141.48-114.498-256-256-256z" />
-<glyph unicode="&#xe927;" glyph-name="toggle-on" horiz-adv-x="1152" d="M1152 448c0-212.078-171.922-384-384-384h-384c-212.078 0-384 171.922-384 384s171.922 384 384 384h384c212.078 0 384-171.922 384-384zM768 704c-141.482 0-256-114.498-256-256 0-141.482 114.498-256 256-256 141.482 0 256 114.498 256 256 0 141.482-114.498 256-256 256z" />
```
Comment (vinyulis) [inline icons/icomoon.svg]: Review deleted icons.

File: src/app/systelab-components/combobox/README.md [hunk 2]
```diff
@@ -144,6 +144,16 @@ public abstract getAllNodeId(): string | number;
 public abstract getAllNodeDescription(): string;
 ```
 
+## Using a Combo with Favourites
+
+In order to create a combobox with favourites you must set the input property "withFavourites" as true, and in your componen add 
+a new service, "PreferencesService", and set it in the **constructor**, and it's necessary to add a **preferenceName** in your component
```
Comment (vinyulis) [inline src/app/systelab-components/combobox/README.md]: preferenceName should be an input

File: src/app/systelab-components/combobox/README.md [hunk 3]
```diff
@@ -177,6 +187,7 @@ Once you have your component, you can use it in your templates.
 | expandToParentContainerHeight | boolean | false ||
 | allowEditInput | boolean | false ||
 | emptyElement | boolean | false ||
+| withFavourites | boolean | false ||
```
Comment (vinyulis) [inline src/app/systelab-components/combobox/README.md]: Add description of the Input
Add preferenceName in the MD

File: src/app/systelab-components/combobox/abstract-combobox.component.ts [hunk 4]
```diff
@@ -254,6 +308,15 @@ export abstract class AbstractComboBox<T> implements AgRendererComponent, OnInit
 		return undefined;
 	}
 
+	protected getComboPreferencesPrefix(): string {
+		return this.preferenceName || this.constructor.name;
+	}
+
+	public setFavourite(event: MouseEvent): void {
```
Comment (vinyulis) [inline src/app/systelab-components/combobox/abstract-combobox.component.ts]: change function name to toggleFavourite

File: src/app/systelab-components/combobox/abstract-combobox.component.ts [hunk 5]
```diff
@@ -268,6 +331,19 @@ export abstract class AbstractComboBox<T> implements AgRendererComponent, OnInit
 		}
 	}
 
+	public addRemoveFavourite(): void {
```
Comment (vinyulis) [inline src/app/systelab-components/combobox/abstract-combobox.component.ts]: Set as private or protected

Comment (vinyulis) [inline src/app/systelab-components/combobox/abstract-combobox.component.ts]: Set name to toggleFavourite

File: src/app/systelab-components/combobox/README.md (hunk 3 above)
Comment (VictorDeBlas) [reply src/app/systelab-components/combobox/README.md]: Done

File: src/app/systelab-components/combobox/abstract-combobox.component.ts (hunk 4 above)
Comment (VictorDeBlas) [reply src/app/systelab-components/combobox/abstract-combobox.component.ts]: Done

File: src/app/systelab-components/combobox/abstract-combobox.component.ts (hunk 5 above)
Comment (VictorDeBlas) [reply src/app/systelab-components/combobox/abstract-combobox.component.ts]: Done

Comment (VictorDeBlas) [reply src/app/systelab-components/combobox/abstract-combobox.component.ts]: Done

File: src/app/systelab-components/combobox/README.md (hunk 2 above)
Comment (VictorDeBlas) [reply src/app/systelab-components/combobox/README.md]: Done

File: src/app/systelab-components/combobox/abstract-combobox.component.ts [hunk 6]
```diff
@@ -268,6 +322,22 @@ export abstract class AbstractComboBox<T> implements AgRendererComponent, OnInit
 		}
 	}
 
+	public addRemoveFavourite(event: MouseEvent): void {
+		event.stopPropagation();
+		jQuery('#' + this.comboId)
```
[the remark this answers is not in the document]
Comment (VictorDeBlas) [reply src/app/systelab-components/combobox/abstract-combobox.component.ts]: Working now

File: icons/icomoon.svg (hunk 1 above)
Comment (VictorDeBlas) [reply icons/icomoon.svg]: Recovered

**Comment by VictorDeBlas** [comment]
<iframe src="https://giphy.com/embed/3ohzdIuqJoo8QdKlnW" width="480" height="222" frameBorder="0" class="giphy-embed" allowFullScreen></iframe><p><a href="https://giphy.com/gifs/reactionseditor-yes-awesome-3ohzdIuqJoo8QdKlnW">via GIPHY</a></p>

**Pull request merged**